src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -138,7 +138,7 @@ private void initRemoteAgent() throws IOException, InterruptedException {
             final SSHUserPrivateKey c = CredentialsProvider.findCredentialById(id, SSHUserPrivateKey.class, build);
             CredentialsProvider.track(build, c);
             if (c == null && !step.isIgnoreMissing()) {
-                listener.fatalError(Messages.SSHAgentBuildWrapper_CredentialsNotFound());
+                throw new IOException(Messages.SSHAgentBuildWrapper_CredentialsNotFound(id));
```

Comment:
`AbortException` would be preferable, as mentioned in Jira, since it avoids printing a stack trace which in this context is not meaningful.

Reply:
I think I did this even though you did mention `AbortException` in Jira since https://github.com/jenkinsci/ssh-agent-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBuildWrapper.java#L207 already exists, is it better to have them raise the same thing or to just raise `AbortException` here?

Reply:
You can correct `SSHAgentBuildWrapper` (used for non-Pipeline projects) if you like, or leave it alone, so long as the Pipeline step does the right thing.